Failure to build due to clockid_t config flag not taking effect

**Description of the bug:**

fio fails to build with the following error. Using modern GCC, I was able to build up to version 3.7 in the same environment.
```
user@home-dt-3 fio % make -j1 V=1
...
crc/../fio_time.h:34:50: error: unknown type name 'clockid_t'; did you mean 'clock_t'?
   34 | extern void set_epoch_time(struct thread_data *, clockid_t, clockid_t);
      |                                                  ^~~~~~~~~
      |                                                  clock_t
crc/../fio_time.h:34:61: error: unknown type name 'clockid_t'; did you mean 'clock_t'?
   34 | extern void set_epoch_time(struct thread_data *, clockid_t, clockid_t);
      |                                                             ^~~~~~~~~
      |                                                             clock_t
```
If my understanding is correct, since 25f8227 this error is not supposed to occur as long as the `clockid_t` configure flag is set to `no`, which it is.
```
user@home-dt-3 fio % ./configure --cc=gcc-14
...
clockid_t                     no
```
**Environment**:  Mac OS X 10.5.9 (Sorbet Leopard) on PowerPC

**fio version**: 3.41

--- a/crc/test.c
+++ b/crc/test.c
@@ -3,10 +3,10 @@
 #include <stdlib.h>
 #include <string.h>
 
+#include "../os/os.h"
 #include "../gettime.h"
 #include "../fio_time.h"
 #include "../lib/rand.h"
-#include "../os/os.h"
 
 #include "../crc/md5.h"
 #include "../crc/crc64.h"

--- a/lib/memcpy.c
+++ b/lib/memcpy.c
@@ -5,9 +5,9 @@
 
 #include "memcpy.h"
 #include "rand.h"
+#include "../os/os.h"
 #include "../fio_time.h"
 #include "../gettime.h"
-#include "../os/os.h"
 
 #define BUF_SIZE	32 * 1024 * 1024ULL
 

